translation_stats/translator_language_proficiency_standard_babel_template.py

```diff
@@ -0,0 +1,113 @@
+import pymysql
+import csv
+import socket
+
+def make_connection(wiki, replica_type="analytics"):
+    """Connects to a host and database of the same name.
+    
+    `replica_type` can be either "analytics" (default) or "web"."""
+    assert replica_type in ("analytics", "web")
+
+    try:
+        connection = pymysql.connect(
+            host=f"{wiki}.{replica_type}.db.svc.wikimedia.cloud",
+            read_default_file="~/.my.cnf",
+            database=f"{wiki}_p",
+            charset='utf8'
+        )
+        return connection
+    except socket.gaierror as e:
+        error_msg = f"Hostname resolution failed: {e}"
+        print(error_msg)
+        raise pymysql.Error(error_msg)
+    except pymysql.Error as e:
+        error_msg = f"Error occurred while connecting to the database: {e}"
+        print(error_msg)
+        raise pymysql.Error(error_msg)
+
+
+def query(conn, query):
+    """Execute a SQL query against the connection, and return **all** the results."""
+    with conn.cursor() as cur:
+        cur.execute(query)
+        data = cur.fetchall()
+        return data
+
+def fetch_babel_data(database):
+    try:
+        conn = make_connection(database)
+        results = query(
+            conn,
+            "SELECT Distinct a.actor_name AS username, b.babel_lang as language_used, b.babel_level as language_level FROM revision r JOIN change_tag ct ON r.rev_id = ct.ct_rev_id JOIN change_tag_def ctd ON ct.ct_tag_id = ctd.ctd_id AND ctd.ctd_name = 'contenttranslation' LEFT JOIN actor a ON r.rev_actor = a.actor_id LEFT JOIN babel b ON b.babel_user = a.actor_user WHERE b.babel_user IS NOT NULL ORDER BY a.actor_name ;"
+        )
+        conn.close()
+        return results
+    except pymysql.OperationalError as e:
```

Comment:
Might be fine to not catch these exceptions and let them crash the process instead.  Is this actually recoverable, or should it be fatal?